File: web/src/layout/WebsocketDemoXml.ts
//
// WebsocketDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './websocket_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.WebsocketDemoXml
export class WebsocketDemoXml {
    xmlRoot!: HTMLElement
    items!: HTMLDivElement
    input!: HTMLInputElement
    submit!: HTMLButtonElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.items = getViewById<HTMLDivElement>(view, "items")
        this.input = getViewById<HTMLInputElement>(view, "input")
        this.submit = getViewById<HTMLButtonElement>(view, "submit")
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/WebsocketDemoXml.kt
//
// WebsocketDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class WebsocketDemoXml {

    lateinit var items: com.lightningkite.butterfly.views.widget.VerticalRecyclerView
    lateinit var input: EditText
    lateinit var submit: ImageButton
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.websocket_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        items = view.findViewById<com.lightningkite.butterfly.views.widget.VerticalRecyclerView>(R.id.items)
        input = view.findViewById<EditText>(R.id.input)
        submit = view.findViewById<ImageButton>(R.id.submit)
        
        
        
        return view
    }
}